pkg/store/gcs.go

```diff
@@ -239,9 +252,23 @@ func (s *GCSStore) loadBlocks() error {
 		b, err := s.loadFromDisk(id)
 		if err != nil {
 			level.Warn(s.logger).Log("msg", "loading block failed", "err", err)
+			continue
 		}
 		s.setBlock(id, b)
+
+		if oldestBlockMinTime > b.meta.MinTime || oldestBlockMinTime == 0 {
+			oldestBlockMinTime = b.meta.MinTime
+		}
+
+		if youngestBlockMaxTime < b.meta.MaxTime || youngestBlockMaxTime == 0 {
+			youngestBlockMaxTime = b.meta.MaxTime
+		}
 	}
+
+	s.oldestBlockMinTime = oldestBlockMinTime
+	s.youngestBlockMaxTime = youngestBlockMaxTime
+	// Update gossip metadata.
+	s.metaUpdater.SetTimestamps(s.oldestBlockMinTime, s.youngestBlockMaxTime)
 	return nil
 }
```

Comment:
Now that I see it, I wonder how much sense the timestamps make for history (? :)) nodes.
The useable value here is the max time, which is a maximum of maximums. This does not tell us anything about whether we already have data up to that timestamp for any specific source node.

So in contrast to source nodes/sidecars, we might not be able to use this interval. We can keep it around for now though. But if we find out it's never going to be used, we should probably drop it.

Reply:
I can see your point. I think it can give us a space for additional optimizations.